File: docs/examples/infoboard.ts
export const processes = {
  infoboard: {
    id: "infoboard",
    handler: async (ctx, msg) => {
      const kv = ctx.kv;
      const ready = await kv.get("ready");

      switch (msg.type) {
        case "get": {
          return ready
            ? {
              name: await kv.get("name"),
              bio: await kv.get("bio"),
              links: await kv.get("links"),
            }
            : { name: "N/A", bio: "Not set up yet." };
        }
        case "setBio": {
          await kv.set("bio", msg.bio);
          break;
        }
        case "setName": {
          await kv.set("name", msg.name);
          break;
        }
        case "setLinks": {
          await kv.set("links", msg.links);
          break;
        }
      }
    },
  },
};

File: docs/examples/infoboard.lua
local processes = {
  infoboard = {
    id = "infoboard",
    handler = function(ctx, msg)
      local kv = ctx.kv
      local ready = kv.get("ready")

      if msg.type == "get" then
        if ready then
          return {
            name = kv.get("name"),
            bio = kv.get("bio"),
            links = kv.get("links"),
          }
        else
          return { name = "N/A", bio = "Not set up yet." }
        end
      elseif msg.type == "setBio" then
        kv.set("bio", msg.bio)
      elseif msg.type == "setName" then
        kv.set("name", msg.name)
      elseif msg.type == "setLinks" then
        kv.set("links", msg.links)
      end
    end,
  },
}

return processes